conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosForkInvalidatingAmender.scala

```diff
@@ -0,0 +1,82 @@
+package tech.cryptonomic.conseil.indexer.tezos
+
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.{BlockLevel, TezosBlockHash}
+import tech.cryptonomic.conseil.indexer.tezos.{TezosDatabaseOperations => DBOps}
+import tech.cryptonomic.conseil.indexer.ForkAmender
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+import slick.dbio.DBIO
+import slick.jdbc.PostgresProfile.api._
+import cats.implicits._
+import slickeffect.implicits._
+
+/** Provides static utilities */
+object TezosForkInvalidatingAmender {
+  def apply(implicit ec: ExecutionContext): TezosForkInvalidatingAmender = new TezosForkInvalidatingAmender
+}
+
+/** Defines the actual logic that amends any data that is identified
+  * on a fork no longer valid.
+  * This implementation will mark such data as invalidated, for all entities
+  * that supports it via fields.
+  * Other data will be simply removed, if no real information loss is implied
+  * E.g. removing processed chain events from the db registry.
+  */
+class TezosForkInvalidatingAmender(implicit ec: ExecutionContext) extends ForkAmender[DBIO, TezosBlockHash] {
+
+  /* Note that we need to defer constraint checks manually with postgres
+   * policies on consistency levels.
+   * Since we need to break constraints between different tables referring
+   * to the blocks being invalidated, until all entries are invalidated consistently,
+   * we tell the db to wait until the transaction session commits before running any verification.
```

Comment:
Since the transaction would violate the constraints, the database will fail to commit, rollback the changes, including the generated fork table entry, and then stop the application.

This means that we should have a guaranteed consistent state of the data as it was right before the detection.